src/server/ua_services_nodemanagement.c

```diff
@@ -1982,6 +1999,19 @@ deleteNodeOperation(UA_Server *server, UA_Session *session, void *context,
         return;
     }
 
+    /* Check the RolePermissions */
+    UA_RoleSet delNodeRoles = node->head.rolePermissions[UA_ROLEPERMISSIONINDEX_DELETENODE];
+    if(session != &server->adminSession &&
+       !UA_RoleSet_intersects(delNodeRoles, session->roles)) {
+        UA_LOG_NODEID_INFO(&node->head.nodeId,
+        UA_LOG_INFO_SESSION(server->config.logging, session, "DeleteNode (%.*s): "
+                            "Cannot delete the node because of missing RolePermisions",
+                            (int)nodeIdStr.length, nodeIdStr.data));
+        UA_NODESTORE_RELEASE(server, node);
+        *result = UA_STATUSCODE_BADUSERACCESSDENIED;
+        return;
+    }
```

Comment:
If I understand this correctly, this checks, if the session has a role assigned which is required to gain the DELETENODE permission. What is the difference of this check and the one in the AccessControl Plugin (allowDeleteNode)?

Reply:
The AccessControl plugin checks go on top of the RolePermissions.
Everything the RolePermissions do can also be done in the pure AccessControl plugin.
But it will be a lot easier to move most checks into the Role-Based Access Control.
Because you don't to build all the machinery to track node access permissions individually.